frontend/src/components/layout/Header/UserProfile.tsx

```diff
@@ -0,0 +1,78 @@
+import useKeycloakWrapper from 'hooks/useKeycloakWrapper';
+import React from 'react';
+import { Image, NavDropdown } from 'react-bootstrap';
+import { useConfiguration } from 'hooks/useConfiguration';
+import profileUrl from 'assets/images/profile.svg';
+import { useHistory } from 'react-router-dom';
+import useCodeLookups from 'hooks/useLookupCodes';
+import styled from 'styled-components';
+import { FaSignOutAlt } from 'react-icons/fa';
+import * as API from 'constants/API';
+import { ILookupCode } from 'actions/lookupActions';
+
+const StyleDropDown = styled(NavDropdown)`
+  font-size: 14px;
+  .dropdown-menu {
+    width: 300px;
+    height: 120px;
+  }
+  .nav-link {
+    color: #fff;
+    padding: 1px;
+  }
+`;
+
+/** Component that allows the user to logout, and gives information on current user's agency/roles */
+export const UserProfile: React.FC = () => {
+  const keycloak = useKeycloakWrapper();
+  const history = useHistory();
+  const displayName =
+    keycloak.displayName ??
+    (!!keycloak.firstName && !!keycloak.lastName
+      ? `${keycloak.firstName} ${keycloak.lastName}`
+      : 'default');
+  const configuration = useConfiguration();
+  const lookupCodes = useCodeLookups();
+  const agencyOptions = lookupCodes.getByType(API.AGENCY_CODE_SET_NAME);
+  const roles = keycloak.roles.join(', ');
+
+  return (
+    <>
+      <Image src={profileUrl} rounded style={{ height: 30, width: 30 }} />
+      <StyleDropDown className="px-0" title={displayName} id="user-dropdown">
+        {history ? (
+          <>
+            <p style={{ margin: 5 }}>
+              <b>
+                {
+                  agencyOptions.find(
+                    (x: ILookupCode) => x.id.toString() === keycloak.agencyId?.toString(),
+                  )?.name
+                }
+              </b>
+            </p>
+            <div style={{ backgroundColor: '#f2f2f2', margin: 5 }}>
```

Comment:
I'd like to stay away from using style as much as this, it makes css inheritance impossible which makes re-use difficult. Prefer either styled components or bootstrap classes.

Reply:
Might not need to rework this, what do you think @goemen ?

Reply:
I can move them into styled components!